### PR TITLE
NNFrames: save load for python

### PR DESCRIPTION
**Pull request by hhbyyh**
provide a prototype, tested manually. Will add some unit tests.

### CODE REVIEW

**Comment by hhbyyh** [comment]
Done. Exposed `nnModel.model`. Please check unit test for the usage.

File: pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py [hunk 1]
```diff
@@ -298,6 +300,47 @@ def test_nnModel_set_Preprocessing(self):
...
+        estimator = NNEstimator(model, criterion).setMaxEpoch(1).setBatchSize(4)
+
+        df = self.get_estimator_df()
+        nnModel = estimator.fit(df)
+        try:
+            tmp_dir = tempfile.mkdtemp()
+            modelPath = os.path.join(tmp_dir, "model")
+            nnModel.model.save(modelPath)
+            loaded_model = Model.load(modelPath)
+            resultDF = NNModel(loaded_model).transform(df)
```
Comment (hhbyyh) [inline pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py]: save/load BigDL model and reconstruct NNModel

**Comment by hhbyyh** [comment]
also cc @yangw1234 for review. Thanks.
Right now this is not supporting save/load within Spark ML Pipeline due to the constraints from Spark. https://issues.apache.org/jira/browse/SPARK-24632 is tracking the issue.

File: pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py (hunk 1 above)
Comment (jason-dai) [reply pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py]: Can we do the same in Scala?

Comment (hhbyyh) [reply pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py]: Yes, Scala has more native support for this.

**Comment by yangw1234** [comment]
LGTM

**Comment by hhbyyh** [comment]
Thanks for the review. @yangw1234 .
@jason-dai let me know if there's other suggestions.

**Comment by jason-dai** [comment]
LGTM

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/709/ passed.
Thanks for reviewing. Merging to master

**Pull request merged**